### ISSUE TITLE
Add client IP for audit logs

### ISSUE THREAD

**Issue by NitroCao** [goharbor/harbor#18675]
Currently audit logs don't contain client IP, which isn't enough for security audit. Can we add it for audit log?

**Comment by chlins** [comment]
Currently, the client IP is not stored in the database, but maybe it can be grabbed from the nginx or core logs.

**Comment by NitroCao** [comment]
> Currently, the client IP is not stored in the database, but maybe it can be grabbed from the nginx or core logs.

Yeah Nginx logs contain client IPs but it's difficult to associate them with corresponding accounts.
I just found related issues:
* #16423 just records the IPs failed to login.
* #5561 is the exact same as my issue, and a related proposal goharbor/community#10 is still open and has no any progress.

**Comment by dioguerra** [comment]
Any news on this...

This is important to track security related issues...

**Comment by reasonerjt** [comment]
I agree having IP information is helpful, but it's secure only when it's accurate.  I'm no expert in networking, but if the client is using VPN or from a different network, getting the source IP may be complicated.

Storing inaccurate information will only have negative effect to security, no?

**Comment by NitroCao** [comment]
> I agree having IP information is helpful, but it's secure only when it's accurate. I'm no expert in networking, but if the client is using VPN or from a different network, getting the source IP may be complicated.

@reasonerjt It's unnecessary for developers to consider such complicated situations, it's for security team's.

**Comment by kofj** [comment]
In our case of cloud's customer, they required record client ip to the audit logs. But it is difficult to obtain the client IP accurately if the user uses a proxy, or the server is deployed in some special network area, such as behind CloudFlare.

**Comment by XANi** [comment]
@kofj 

Usual way to do it is:

* have array with network whitelist ( say 127.0.0.0/8 by default or some docker IPs ,so proxying from local HAProxy/Nginx works without setup)
* only accept `X-Forwarded-For` from IPs from that whitelist

Then the operator can just add a network/IP of loadbalancer(s) into that and it will only accept the  header from there, and then it is up to loadbalancer config to reject the XFF from the world and insert its own.

So on Harbor side it would be just network whitelist + name of the header (XFF is common, so is x-real-ip from nginx so config here is required)
